src/render/mod.rs

```diff
@@ -953,6 +939,109 @@ impl Renderer {
         self.encoder.flush(&mut self.device);
     }
 
+    #[inline]
+    fn instance_basic(
```

Comment:
maybe have those constructors inside `impl Instance`?

Reply:
yes, it'd be better